src/contrib/cluster/Akka.Cluster.Sharding.Tests/WrappedShardBufferedMessageSpec.cs

```diff
@@ -0,0 +1,250 @@
+
+        private readonly string _shardId;
+        private readonly IActorRef _probe;
+
+        public FakeShardStoreActor(string shardId, IActorRef probe)
+        {
+            _shardId = shardId;
+            _probe = probe;
+            Context.System.EventStream.Publish(new ShardStoreCreated(Self, shardId));
+        }
+
+        protected override bool Receive(object message)
+        {
+            switch (message)
+            {
+                case RememberEntitiesShardStore.GetEntities:
+                    Sender.Tell(new RememberEntitiesShardStore.RememberedEntities(ImmutableHashSet<string>.Empty));
+                    return true;
+                case RememberEntitiesShardStore.Update m:
+                    _probe.Tell(new RememberEntitiesShardStore.UpdateDone(m.Started, m.Stopped));
+                    return true;
+            }
+            return false;
+        }
+    }
+
+    private class FakeCoordinatorStoreActor : ActorBase
+    {
+        public static Props Props() => Actor.Props.Create(() => new FakeCoordinatorStoreActor());
+
+        public FakeCoordinatorStoreActor()
+        {
+            Context.System.EventStream.Publish(new CoordinatorStoreCreated(Context.Self));
+        }
+
+        protected override bool Receive(object message)
+        {
+            switch (message)
+            {
+                case RememberEntitiesCoordinatorStore.GetShards _:
+                    Sender.Tell(new RememberEntitiesCoordinatorStore.RememberedShards(ImmutableHashSet<string>.Empty));
+                    return true;
+                case RememberEntitiesCoordinatorStore.AddShard m:
+                    Sender.Tell(new RememberEntitiesCoordinatorStore.UpdateDone(m.ShardId));
+                    return true;
+            }
+            return false;
+        }
+    }
+
+    private static Config GetConfig()
+    {
+        return ConfigurationFactory.ParseString(@"
+                akka.loglevel=DEBUG
+                akka.actor.provider = cluster
+                akka.remote.dot-netty.tcp.port = 0
+                akka.cluster.sharding.state-store-mode = ddata
+                akka.cluster.sharding.remember-entities = on
+
+                # no leaks between test runs thank you
+                akka.cluster.sharding.distributed-data.durable.keys = []
+                akka.cluster.sharding.verbose-debug-logging = on
+                akka.cluster.sharding.fail-on-invalid-entity-state-transition = on")
+
+            .WithFallback(Sharding.ClusterSharding.DefaultConfig())
+            .WithFallback(DistributedData.DistributedData.DefaultConfig())
+            .WithFallback(ClusterSingleton.DefaultConfig());
+    }
+    
+    private class MessageExtractor: HashCodeMessageExtractor
+    {
+        public MessageExtractor() : base(10)
+        {
+        }
+
+        public override string EntityId(object message)
+        {
+            return message switch
+            {
+                Message m => m.Payload,
```

Comment:
This is kind of testing the wrong thing here - the `Unapply` method should be handling this.

Reply:
no its not, we're testing `MyEnvelope`, that `Message` class is the actual message being sent, not a `string`.

Reply:
actually, you're right, there's something wrong with the repro code